UI/TraceDataLinux.cpp

```diff
@@ -0,0 +1,316 @@
+    {
+      return l.m_Entry->m_ImageBase < r.m_Entry->m_ImageBase;
+    });
+
+
+  int resolve_count = 0;
+  int fail_count = 0;
+
+  QSet<uintptr_t> ripLookup;  
+  const auto& addRipToModuleFrames = [&](uintptr_t rip) -> void
+  {
+    if ( ripLookup.contains(rip) )
+    {
+      return;
+    }
+    ripLookup.insert(rip);
+
+    for ( int i = 0; i < moduleCount; i++ )
+    {
+      ModuleFrames& module = moduleFrameList[i];
+      if ( rip >= module.m_Entry->m_ImageBase + module.m_Entry->m_ImageSegmentOffset)
+      {
+        if ( rip < (module.m_Entry->m_ImageBase + module.m_Entry->m_ImageSegmentOffset + module.m_Entry->m_SizeBytes) )
+        {
+          module.m_Frames.push_back(rip);
+          break;
+        }
+        else
+          continue;
+      }
+      else
+      {
+        qDebug() << "Failed to find module for address " << rip;
+        fail_count++;
+      }
+    }
+  };
+
+  
+  int total = 2 * (stackCount + nodeCount); // Two passes, once to sort the data and once to process it
+  int completed = 0;
+
+  // Sort instructions in stacks
+  for (uint32_t i = 0; i < stackCount; ++i, ++completed)
+  {
+    if (uintptr_t rip = stacks[i])
+    {
+      addRipToModuleFrames(rip);
+    }
+
+    if (0 == (completed & 0x400))
+    {
+      Q_EMIT symbolResolutionProgressed(completed, total);
+    }
+  }
+
+  // Sort any instructions used in leaf functions.
+  for (uint32_t i = 0; i < nodeCount; ++i, ++completed)
+  {
+    addRipToModuleFrames(nodes[i].m_Rip);
+
+    if (0 == (completed & 0x400))
+    {
+      Q_EMIT symbolResolutionProgressed(completed, total);
+    }
+  }
+
+  Q_EMIT symbolResolutionProgressed(completed, total);
+
+  QVector<SerializedSymbol> symbols;
+
+  QVector<QChar> stringData;
+  stringData.push_back(QChar(0));    // Zero offset strings point here.
+
+  QHash<QString, uint32_t> stringLookup;
+
+  auto intern_qstring = [&](const QString& s) -> uint32_t
+  {
+    auto it = stringLookup.find(s);
+    if (it != stringLookup.end())
+    {
+      return it.value();
+    }
+    uint32_t result = stringData.size();
+    Q_FOREACH (QChar ch, s)
+    {
+      stringData.append(ch);
+    }
+    stringData.append(QChar(0));
+    return result;
+  };
+
+  auto intern_string = [&](const char* str) -> uint32_t
+  {
+    return intern_qstring(QString::fromUtf8(str));
+  };
+
+  for ( int i = 0; i < moduleFrameList.count(); i++ )
+  {
+    if ( moduleFrameList[i].m_Frames.count() == 0 )
+    {
+        continue;
+    }
+
+    // Figure out the actual file with the debug info. If there's anything in /usr/lib/debug, use that version instead
+    const char* baseFilename = hdr->GetModuleName(*moduleFrameList[i].m_Entry);
+    // resolve any symlinks
+    char resolvedBaseFilename[1024];
+    if ( realpath(baseFilename, resolvedBaseFilename) == nullptr )
+    {
+      strcpy(resolvedBaseFilename, baseFilename);
+    }
+    if (strlen(baseFilename)  == 0)
```

Comment:
Ach, yes, thanks. This is what I get for trying to push this before leaving the office late on a friday :)